sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -52,120 +53,124 @@ object ScalaNativePluginInternal {
-      compileLl(clang, target, appll, binary, links, linkage, clangOpts)
-
-      binary
-    },
-
-    run := {
-      val binary = nativeLink.value
-
-      Process(abs(binary)).!
-    }
+      libraryDependencies ++= Seq(
+          "org.scala-native" %% "nativelib" % nativeVersion,
+          "org.scala-native" %% "javalib"   % nativeVersion,
+          "org.scala-native" %% "scalalib"  % nativeVersion
+      ),
+      addCompilerPlugin(
+          "org.scala-native" % "nscplugin" % nativeVersion cross CrossVersion.full),
+      resolvers += Resolver.sonatypeRepo("snapshots"),
+      nativeVerbose := false,
+      nativeClang := {
+        val binaryName = "clang++"
+        val binaryNames = Seq(("3", "8"), ("3", "7")).flatMap {
+          case (major, minor) =>
+            Seq(s"$binaryName$major$minor", s"$binaryName-$major.$minor")
+        } :+ binaryName
+
+        Process("which" +: binaryNames).lines_!
+          .map(file(_))
+          .headOption
+          .getOrElse {
+            throw new MessageOnlyException(
+                s"no ${binaryNames.mkString(", ")} found in $$PATH. Install clang")
+          }
+      },
+      nativeClangOptions := {
+        val includes = ("/usr/local/include" #:: Try(
+                Process("llvm-config --includedir").lines_!)
+              .getOrElse(Stream.empty)).map(s => s"-I$s")
+        val libs = ("/usr/local/lib" #:: Try(
+                Process("llvm-config --includedir").lines_!)
+              .getOrElse(Stream.empty)).map(s => s"-L$s")
+        includes #::: libs
+      },
+      nativeEmitDependencyGraphPath := None,
+      nativeLibraryLinkage := Map(),
+      artifactPath in nativeLink := {
+        (crossTarget in Compile).value / (moduleName.value + "-out")
+      },
+      nativeLink := {
+        val entry = (mainClass in Compile).value.get.toString + "$"
+        val classpath =
+          cpToStrings((fullClasspath in Compile).value.map(_.data))
+        val target    = (crossTarget in Compile).value
+        val appll     = target / (moduleName.value + "-out.ll")
+        val binary    = (artifactPath in nativeLink).value
+        val verbose   = nativeVerbose.value
+        val clang     = nativeClang.value
+        val clangOpts = nativeClangOptions.value
+        val dotpath   = nativeEmitDependencyGraphPath.value
+        val linkage   = nativeLibraryLinkage.value
+        val opts = new NativeOpts(classpath,
+                                  abs(appll),
+                                  dotpath.map(abs),
+                                  entry,
+                                  verbose)
+
+        checkThatClangIsRecentEnough(clang)
+
+        IO.createDirectory(target)
+        unpackRtlib(classpath)
+        val links = compileNir(opts).map(_.name)
+        compileLl(clang, target, appll, binary, links, linkage, clangOpts)
+
+        binary
+      },
+      run := {
+        val binary = nativeLink.value
+
+        Process(abs(binary)).!
+      }
```

Comment:
This block of changes seems to have too big of a diff for no good reason. Please avoid changing the parts of the codebase that are not related to the main topic of the pr. For example removing blank lines between settings was not necessary.